atef/widgets/config/find_replace.py

```diff
@@ -0,0 +1,1009 @@
+
+    def update_change_list(self) -> None:
+        """
+        update the global change list, gathering all ``FindReplaceAction``'s
+        """
+        # walk through edits_table, gather list of list of paths
+        self.all_actions = []
+        for row_idx in range(self.edits_table.rowCount()):
+            template_widget = self.edits_table.cellWidget(row_idx, 0)
+            self.all_actions.extend(template_widget.get_actions())
+
+        self.update_title()
+
+    def show_changes_from_edit(self, *args, **kwargs) -> None:
+        """
+        Populate the details_list with each action from the selected edit.
+        Provide helpful text if the edit is incomplete or invalid.
+        """
+        self.details_list.clear()
+        # on selected callback, populate details table
+        selected_ranges = self.edits_table.selectedRanges()
+        if not selected_ranges:
+            return
+
+        edit_row_widget: TemplateEditRowWidget = self.edits_table.cellWidget(
+            selected_ranges[0].topRow(), 0
+        )
+
+        if not isinstance(edit_row_widget, TemplateEditRowWidget):
+            # placeholder text if nothing is selected
+            l_item = QtWidgets.QListWidgetItem("select an edit or click an edit's"
+                                               "refresh button to show details.")
+            return
+        elif not edit_row_widget.get_details_rows():
+            l_item = QtWidgets.QListWidgetItem(
+                'provide search and replace text to show details...'
+            )
+            self.details_list.addItem(l_item)
+            return
+
+        row_widget: FindReplaceRow
+        for row_widget in edit_row_widget.get_details_rows():
+            l_item = QtWidgets.QListWidgetItem()
+            l_item.setSizeHint(QtCore.QSize(row_widget.width(), row_widget.height()))
+            self.details_list.addItem(l_item)
+            self.details_list.setItemWidget(l_item, row_widget)
+
+            row_widget.remove_item.connect(partial(self.remove_item_from_details, l_item))
+
+    def remove_item_from_details(self, item: QtWidgets.QListWidgetItem) -> None:
+        """ remove an item from the details list """
+        self.details_list.takeItem(self.details_list.row(item))
+
+
+class TemplateEditRowWidget(DesignerDisplay, QtWidgets.QWidget):
+    """
+    A widget for specifying the information for find/replace actions
+
+    Each edit can correspond to multiple ``FindReplaceAction``'s
+    Generates ``FindReplaceRow``'s for each ``FindReplaceAction``
+    """
+    button_box: QtWidgets.QDialogButtonBox
+    child_button: QtWidgets.QPushButton
+
+    setting_button: QtWidgets.QToolButton
+    regex_button: QtWidgets.QToolButton
+    case_button: QtWidgets.QToolButton
+
+    search_edit: QtWidgets.QLineEdit
+    replace_edit: QtWidgets.QLineEdit
+
+    filename = 'template_edit_row_widget.ui'
+
+    def __init__(
+        self,
+        *args,
+        data=None,
```

Comment:
`data` unused?

Reply:
My `TableWidgetWithAddRow` expected its row widgets to be `DataWidget`s, all of which have a `data` kwarg.  I probably could have made a dataclass to represent an "Edit" but apparently past me didn't feel like it 🤷